test/libyul/Parser.cpp

```diff
@@ -306,33 +306,35 @@ BOOST_AUTO_TEST_CASE(if_statement_invalid)
 
 BOOST_AUTO_TEST_CASE(builtins_parser)
 {
-	struct SimpleBuiltins: public Builtins
+	struct SimpleDialect: public Dialect
 	{
-		BuiltinFunction const* query(YulString _name) const override
+		SimpleDialect(): Dialect(AsmFlavour::Strict) {}
+		BuiltinFunction const* builtin(YulString _name) const override
 		{
 			return _name == YulString{"builtin"} ? &f : nullptr;
 		}
 		BuiltinFunction f;
 	};
 
-	Dialect dialect(AsmFlavour::Strict, make_shared<SimpleBuiltins>());
+	shared_ptr<Dialect> dialect = make_shared<SimpleDialect>();
 	CHECK_ERROR_DIALECT("{ let builtin := 6 }", ParserError, "Cannot use builtin function name \"builtin\" as identifier name.", dialect);
 	CHECK_ERROR_DIALECT("{ function builtin() {} }", ParserError, "Cannot use builtin function name \"builtin\" as identifier name.", dialect);
 	CHECK_ERROR_DIALECT("{ builtin := 6 }", ParserError, "Cannot assign to builtin function \"builtin\".", dialect);
 }
 
 BOOST_AUTO_TEST_CASE(builtins_analysis)
 {
-	struct SimpleBuiltinsAnalysis: public Builtins
+	struct SimpleDialectAnalysis: public Dialect
```

Comment:
This struct itself doesn't analyze much, does it? Maybe it should just be called ``SimpleDialect``, same as above?